sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -918,4 +1043,26 @@ public static void ResetFeatureSwitch()
                 "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE");
         }
     }
+#else
+#pragma warning disable SA1507 // File can not contain multiple types
```

Comment:
nit: would it make sense to make this an internal class?  I don't have full context here, so feel free to ignore if not.

Reply:
Ah, nevermind, I see this is just a dupe of existing code.